### ISSUE TITLE
Distored 8-bit output (both signed and unsigned)

### ISSUE THREAD

**Issue by Wohlstand** [Wohlstand/libADLMIDI#82]
@jpcima , two days ago I have found that 8bit output is distorted (on SDL side it's valid, it can play any 8-bit sound validly).

For convenience, I have made the ability use any different sample format from ADLMIDIPlay tool:
https://github.com/Wohlstand/libADLMIDI/commit/8077db2dda93a18116ea48344b5480aef08f59fa

I'll try to fix something myself, but maybe you know a true reason of this?

EDIT: Just now I have tried - unsigned 8-bit output works fine! The problem is on side of signed

**Comment by Wohlstand** [comment]
Okay, just not I have found reason of crap is that output format was opened different from requested:
![default](https://user-images.githubusercontent.com/6751442/39407234-344069b2-4bcb-11e8-8ab3-54176c737682.png)

**Comment by Wohlstand** [comment]
Same happens on attempt to open U16 output:
![default](https://user-images.githubusercontent.com/6751442/39407239-4fdf70e6-4bcb-11e8-9eb2-4576fa2a670b.png)

**Comment by Wohlstand** [comment]
Okay, I have fixed that, I have avoided using of unsupported sample formats by analyzing of `obtained` than using `wanted` format.

**Comment by jpcima** [comment]
Good that you have found the problem. I am not of a lot of help now. (I'm not in super health now.. still resting a bit.)

You're trying to test the sample formats. For ones last added I just checked for few values printed on the console and no more; they seemed to be right. If this should be tested in unit test form, it should be checked at middle value and lower/upper bounds + slightly over bounds for clipping checks.

Upon reading your code there's a little things to comment about. 1. use endian neutral SDL formats (with `SYS` suffix) 2. apply `SDL_AUDIO_MASK_DATATYPE` on the switch value of format

I've progressed opl3 math a bit on my side but not a lot. I think the key-on part is figured out, but still need to test, and repeat the same process on the other one. (take samples of release stage and find a 2 parameter curve fit formula, etc)
